src/extractor.ts

```diff
@@ -43,17 +43,53 @@ export class Extractor {
 	/**
 	 * Serialize and return output
 	 */
-	public serialize(): string {
-		return this.serializer.serialize(this.messages);
+	public serialize(messages: {[key: string]: string}): string {
+		return this.serializer.serialize(messages);
+	}
+
+	/**
+	 * Parse the content of a file and return output as an object
+	 */
+	public parse(contents: string): {[key: string]: string} {
+		return this.serializer.parse(contents);
 	}
 
 	/**
 	 * Serialize and save to destination
 	 */
-	public save(destination: string): string {
-		const data = this.serialize();
-		fs.writeFileSync(destination, data);
-		return data;
+	public save(destination: string, replace: boolean = false, clean: boolean = true): string {
+		let extractedData = this._prepareMessages(this.messages),
+			output: string;
+
+		if (!replace && fs.existsSync(destination)) {
+			const fileData = this.parse(fs.readFileSync(destination, 'utf8'));
+
+			extractedData = this._merge(fileData, extractedData, clean);
+		}
+
+		output = this.serialize(extractedData);
+		fs.writeFileSync(destination, output);
+		return output;
+	}
+
+	protected _prepareMessages(messages: string[]): {[key: string]: string} {
+		let result = {};
+
+		messages.forEach(message => {
+			result[message] = '';
+		});
+
+		return result;
+	}
+
+	protected _merge(fileData: {[key: string]: string}, extractedData: {[key: string]: string}, clean: boolean = true): {[key: string]: string} {
```

Comment:
What do you think about adding a `MessageInterface`?

`{ [key: string]: string }` is used quite a few places right now and makes the code look more complex than it is.

I can do this later if you want to focus on other things.